### ISSUE TITLE
outDir is not working when using tsconfig

### ISSUE THREAD

**Issue by gilamran** [TypeStrong/grunt-ts#312]
This is my grunt config:
```
		ts: {
			options: {
				target: 'es5',
				sourceMap: false,
				declaration: false,
				removeComments: false,
				experimentalDecorators: true,
				module: 'commonjs',
				verbose: true
			},
			build: {
				outDir: '.tmp',
				tsconfig: {
					tsconfig: 'app/tsconfig.json',
					ignoreFiles: true,
					ignoreSettings: false,
					overwriteFilesGlob: false,
					updateFiles: true,
					passThrough: false
				}
			}
		}
```
And my **tsconfig.json** contains only `filesGlob` (No `compilerOptions` section)
Everything works, except that it will not respect the `outDir` option, all the js file are being generated right next to the ts files...
If I add `compilerOptions` with `outDir` inside the **tsconfig.json**, it will respect it... I also tried setting the `ignoreSettings` to true, but it didn't change anything

**Comment by nycdotnet** [comment]
I looked into this tonight.  I believe it is a bug.  I have to think about the correct way to fix it, but it should be doable.  Thanks for the report.

**Comment by stefankolb** [comment]
I'm experiencing a similar issue.

This is my **tsconfig.json**:
```
{
  "compilerOptions": {
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "module": "system",
    "moduleResolution": "node",
    "noImplicitAny": false,
    "outDir": "build/develop/assets/scripts",
    "removeComments": false,
    "rootDir": "src/scripts",
    "sourceMap": false,
    "target": "ES5"
  },
  "exclude": [
    "node_modules"
  ]
}
```

And this is the grunt-ts config in my **Gruntfile.js**:
```
  ts: {
      
      develop: {
        tsconfig: true,
        src: '<%= files_internal.scripts %>'
      }
      
    }
```

**Problem**: When I run grunt, the outDir is respected and .js-files are generated in _build/develop/assets/scripts_, but when I use grunt-contrib-watch execute the ts task, the outDir is not respected and .js-files are genereated right next to their .ts-files.

Here is the grunt-contrib-watch part of my Gruntfile.js:
```
  ts: {
        files: [
          'src/scripts/**/*.ts'
        ],
        tasks: [
          'ts:develop',
          'karma:unit',
          'index:develop'
        ]
      }
```


Any thoughts?

**Comment by nycdotnet** [comment]
Please check out the latest beta which should resolve this issue.

**Comment by stefankolb** [comment]
I'm using 5.3.0-beta.3, is this the latest beta?

**Comment by nycdotnet** [comment]
@stefankolb yes.  Would you please start grunt with the `--verbose` switch and let me know what is being passed to tsc in the different scenarios?  We don't have any tests setup with grunt-config-watch so I'm curious what is different?

**Comment by nycdotnet** [comment]
btw I'm just about to release 5.3.0-beta.4 but it is unlikely to make a difference as it just changes some stuff with HTML template processing

**Comment by stefankolb** [comment]
@nycdotnet Sorry my answer took a while, but I'm quite puzzled. After starting grunt with `--verbose`everything was working fine. So I tried starting grunt without `--verbose` and everything was working fine, too. To be honest, I have no idea what I did to make this work correctly. Sorry to not be of any more help here.

Anyway, here is the output of both scenarios. Maybe this is of some use for you or anyone in the future.

**grunt develop**
```
[...]
Running "ts:develop" (ts) task
Verifying property ts.develop exists in config...OK
Files: src/scripts/boot.ts, src/scripts/app.component.ts
Compiling...
Cleared fast compile cache for target: develop
### Fast Compile >>src/scripts/boot.ts
### Fast Compile >>src/scripts/app.component.ts
Using tsc v1.7.3
"/Users/stefan/development/source/web2epub/src/scripts/boot.ts" "/Users/stefan/development/source/web2epub/src/scripts/app.component.ts" "/Users/stefan/development/source/web2epub/src/scripts/.baseDir.ts" --emitDecoratorMetadata --experimentalDecorators --moduleResolution node --rootDir src/scripts --target ES5 --module system --outDir build/develop/assets/scripts/app
Writing .tscache/develop/timestamp...OK
Writing .tscache/develop/hashes/boot.ts-d4b0f35f1d1c846332e093ec486a9efe...OK
Writing .tscache/develop/hashes/app.component.ts-7d6b337679034f3111d154075e254bac...OK
Writing .tscache/develop/hashes/.baseDir.ts-b06a9234b92109037dbbd5e0f375ed75...OK
[...]
```

**grunt watch** (grunt-contrib-watch), file `boot.ts` has been edited:
```
[...]
Running "ts:develop" (ts) task
Verifying property ts.develop exists in config...OK
Files: src/scripts/boot.ts, src/scripts/app.component.ts
Compiling...
### Fast Compile >>src/scripts/boot.ts
Using tsc v1.7.3
"/Users/stefan/development/source/web2epub/src/scripts/boot.ts" "/Users/stefan/development/source/web2epub/src/scripts/.baseDir.ts" --emitDecoratorMetadata --experimentalDecorators --moduleResolution node --rootDir src/scripts --target ES5 --module system --outDir build/develop/assets/scripts/app
Writing .tscache/develop/timestamp...OK
Writing .tscache/develop/hashes/boot.ts-d4b0f35f1d1c846332e093ec486a9efe...OK
Writing .tscache/develop/hashes/.baseDir.ts-b06a9234b92109037dbbd5e0f375ed75...OK
[...]
```

**Comment by nycdotnet** [comment]
This is resolved via a warning in grunt-ts v5.3.0 which was just pushed to npm. Please comment or re-open if there are any remaining issues. Thank you for contributing!

I appreciate the info @stefankolb - I am assuming it's resolved now. Please let me know if it happens again with 5.3.0 final.
